File: android-aop-apt/src/main/java/com/flyjingfish/android_aop_processor/CollectMethod.java
package com.flyjingfish.android_aop_processor;

public class CollectMethod {
    /**
     * 需要收集的类名（包含包名）
     */
    String collectClassName;
    /**
     * 执行类名（包含包名）
     */
    String invokeClassName;
    /**
     * 执行方法
     */
    String invokeMethod;
    /**
     * 是否是 class
     */
    String isClazz;
    /**
     * 正则表达式
     */
    String regex;
    /**
     * 收集的类型
     */
    String collectType;

    public CollectMethod(String collectClassName, String invokeClassName, String invokeMethod, String isClazz, String regex, String collectType) {
        this.collectClassName = collectClassName;
        this.invokeClassName = invokeClassName;
        this.invokeMethod = invokeMethod;
        this.isClazz = isClazz;
        this.regex = regex;
        this.collectType = collectType;
    }
}

File: android-aop-ksp/src/main/java/com/flyjingfish/android_aop_ksp/CollectMethod.kt
package com.flyjingfish.android_aop_ksp

import com.google.devtools.ksp.symbol.KSFunctionDeclaration

data class CollectMethod(
    val symbol : KSFunctionDeclaration,
    /**
     * 需要收集的类名（包含包名）
     */
    val collectClassName: String,
    /**
     * 执行类名（包含包名）
     */
    val invokeClassName: String,
    /**
     * 执行方法
     */
    val invokeMethod: String,
    /**
     * 是否是 class
     */
    val isClazz: String = "false",
    /**
     * 正则表达式
     */
    val regex: String = "",
    /**
     * 收集的类型
     */
    val collectType: String
)